sbt-converter/src/main/scala/org/scalablytyped/converter/internal/PluginRemoteCache.scala

```diff
@@ -47,6 +51,63 @@ object PluginRemoteCache {
             }
           }
           logger.warn(s"pushed to $pushLocation")
+
+        case RemoteCache.S3(pushConfiguration, _) =>
+          val (runCachePath, output) = inTask.value
+          val logger                 = WrapSbtLogger.task.value
+
+          val s3 = {
+            implicit class RichBuilder(s3ClientBuilder: S3ClientBuilder) {
+              def add[T](value: Option[T], builder: S3ClientBuilder => T => S3ClientBuilder) =
+                value.fold(s3ClientBuilder)(builder(s3ClientBuilder))
+            }
+
+            S3Client
+              .builder()
+              .add(pushConfiguration.region, _.region)
+              .add(pushConfiguration.endpointOverride, _.endpointOverride)
+              .add(pushConfiguration.credentialsProvider, _.credentialsProvider)
+              .build()
+          }
+
+          def upload(relPath: RelPath) = {
+            val key = s"${pushConfiguration.prefix.getOrElse("")}/${relPath.toString()}"
+
+            val exists = Try(
+              s3.headObject(
+                HeadObjectRequest
+                  .builder()
+                  .bucket(pushConfiguration.bucket)
+                  .key(key)
+                  .build(),
+              ),
+            ).isSuccess
+
+            if (exists) {
+              logger.warn(s"skipping already cached file $relPath")
+            } else {
+              logger.warn(s"uploading $relPath")
+              s3.putObject(
+                PutObjectRequest
+                  .builder()
+                  .bucket(pushConfiguration.bucket)
+                  .key(key)
+                  .build(),
+                (Utils.IvyLocal.value / relPath).toNIO,
+              )
+            }
+          }
+
+          Lock.synchronized {
+            locally {
+              output.allRelPaths.foreach(upload)
+            }
+            locally {
+              val relPath: os.RelPath = runCachePath.relativeTo(Utils.IvyLocal.value)
```

Comment:
I don't think this is correct, it's stored within `(Global / stDir).value`. 

All this path computation probably made this all rather confusing. Just to clear things up, this is what it looks like on the server for the rsync implementation (i'd rather we use the same layout for s3):
```
$ find st-cache/
st-cache/org.scalablytyped/ant-design__icons_sjs1_2.13/4.2.1-5fabe8/jars
st-cache/org.scalablytyped/ant-design__icons_sjs1_2.13/4.2.1-5fabe8/jars/.lock
st-cache/org.scalablytyped/ant-design__icons_sjs1_2.13/4.2.1-5fabe8/jars/ant-design__icons_sjs1_2.13.jar
st-cache/org.scalablytyped/ant-design__icons_sjs1_2.13/4.2.1-5fabe8/ivys
st-cache/org.scalablytyped/ant-design__icons_sjs1_2.13/4.2.1-5fabe8/ivys/ivy.xml
st-cache/org.scalablytyped/ant-design__icons_sjs1_2.13/4.2.1-5fabe8/poms
st-cache/org.scalablytyped/ant-design__icons_sjs1_2.13/4.2.1-5fabe8/poms/ant-design__icons_sjs1_2.13.pom
st-cache/org.scalablytyped/ant-design__icons_sjs1_2.13/4.2.1-5fabe8/srcs
st-cache/org.scalablytyped/ant-design__icons_sjs1_2.13/4.2.1-5fabe8/srcs/ant-design__icons_sjs1_2.13-sources.jar
st-cache/org.scalablytyped/ant-design__icons_sjs1_2.13/4.2.1-2b8648
st-cache/org.scalablytyped/ant-design__icons_sjs1_2.13/4.2.1-2b8648/jars
st-cache/org.scalablytyped/ant-design__icons_sjs1_2.13/4.2.1-2b8648/jars/ant-design__icons_sjs1_2.13.jar
st-cache/org.scalablytyped/ant-design__icons_sjs1_2.13/4.2.1-2b8648/ivys
st-cache/org.scalablytyped/ant-design__icons_sjs1_2.13/4.2.1-2b8648/ivys/ivy.xml
st-cache/org.scalablytyped/ant-design__icons_sjs1_2.13/4.2.1-2b8648/poms
st-cache/org.scalablytyped/ant-design__icons_sjs1_2.13/4.2.1-2b8648/poms/ant-design__icons_sjs1_2.13.pom
st-cache/org.scalablytyped/ant-design__icons_sjs1_2.13/4.2.1-2b8648/srcs
st-cache/org.scalablytyped/ant-design__icons_sjs1_2.13/4.2.1-2b8648/srcs/ant-design__icons_sjs1_2.13-sources.jar
st-cache/runs
st-cache/runs/a40475394e857f2b2e09403e98e8f938.json
st-cache/runs/db82c7fa53ea4f31d03448802ffcabd1.json
st-cache/runs/b99000f804d5ea41c1fef6b89fb793d3.json
st-cache/runs/3fca675b2fa8a7c00829d9e32d376c01.json
```